solarforecastarbiter/io/reference_observations/reference_data.py

```diff
@@ -0,0 +1,319 @@
+                   'UO SRML', 'SANDIA']
+
+DEFAULT_SITEFILE = resource_filename(
+    Requirement.parse('solarforecastarbiter'),
+    'solarforecastarbiter/io/reference_observations/sfa_reference_sites.csv')
+
+
+SFA_REFERENCE_TOKEN = os.getenv('SFA_REFERENCE_TOKEN')
+SFA_API_BASE_URL = os.getenv('SFA_API_BASE_URL')
+
+logger = logging.getLogger('reference_data')
+
+CLI_DESCRIPTION = """
+Tool for initializing and updating SolarForecastArbiter reference observation data.
+Supports importing sites from the following networks:
+
+NOAA (The National Oceanic and Atmospheric Administration)
+    SURFRAD: Surface Radiation Budget Network
+    https://www.esrl.noaa.gov/gmd/grad/surfrad/
+
+    SOLRAD:
+    https://www.esrl.noaa.gov/gmd/grad/solrad/index.html
+
+    CRN: U.S. Climate Reference Network
+    https://www.ncdc.noaa.gov/crn/
+
+NREL MIDC: National Renewable Energy Laboratory Measurement and Instrumentation Data Center
+    https://midcdmz.nrel.gov/
+
+UO SRML: University of Oregon Solar Radiation Monitoring Laboratory
+    http://solardat.uoregon.edu/
+
+SANDIA: Sandia National Laboratory Regional Test Centers for Solar Technologies
+    https://pv-dashboard.sandia.gov/
+"""  # noqa: E501
+
+
+def add_network_arg(parser):
+    """Adds the `--networks` argument to a subparser so that it does not consume
+    the the init and update.
+    """
+    parser.add_argument(
+        '--networks', nargs='+', default=NETWORK_OPTIONS,
+        choices=NETWORK_OPTIONS,
+        help="The Networks to act on. Defaults to all.")
+
+
+def get_apisession():
+    return APISession(SFA_REFERENCE_TOKEN, base_url=SFA_API_BASE_URL)
+
+
+def create_site(api, site):
+    """Post a new site to the API and create it's applicable reference
+    observations.
+
+    Parameters
+    ----------
+    api : io.APISession
+        An APISession with a valid JWT for accessing the Reference Data user.
+    site : dict
+        Dictionary describing the site to post. This will be instantiated as
+        a datamodel.Site object and the value of 'extra_parameters' will be
+        serialized to json.
+
+    Returns
+    -------
+    uuid : string
+        UUID of the created Site.
+    """
+    # get a reference to network before we serialize extra_parameters
+    network = site['extra_parameters']['network']
+    site.update({'extra_parameters': json.dumps(site['extra_parameters'])})
+    site_name = common.clean_name(site['name'])
+    site['name'] = f'{network} {site_name}'
+    site_to_create = Site.from_dict(site)
+    try:
+        created = api.create_site(site_to_create)
+    except HTTPError as e:
+        logger.error(f"Failed to create Site {site['name']}.")
+        logger.debug(f'HTTP Error: {e.response.text}')
+    else:
+        logger.info(f'Created Site {created.name} successfully.')
+        network_handler = NETWORKHANDLER_MAP.get(network)
+        if network_handler is None:
+            logger.warning(f'Unrecognized network, {network} on Site '
```

Comment:
given that this is part of reference_data, should this check happen before making the site adn then fail